lib/str-utils.h

```diff
@@ -71,4 +72,29 @@ void g_string_steal(GString *s);
 
 gchar *__normalize_key(const gchar* buffer);
 
+
+/* This version of strchr() is optimized for cases where the string we are
+ * looking up characters in is often zero or one character in length.  In
+ * those cases we can avoid the strchr() call, which can make a tight loop
+ * doing a lot of strchr() calls a lot faster, especially as this function
+ * is inlined.
+ *
+ * Naming: I originally wanted to use the original function as a prefix
+ * (strchr), however that would potentially pollute the namespace of the
+ * library that we are optimizing.  So I've added an underscore prefix in
+ * order not to clash, with that it is still obvious that it intends to
+ * behave the same as strchr().
+ *
+ * NOTE: don't use this unless strchr() really shows up in your profile.
+ */
+static inline char *
```

Comment:
Because I wanted it to match the strchr() prototype.
On Jan 26, 2016 3:40 PM, "Tamas Nagy" <notifications@github.com> wrote:

> In lib/str-utils.h
> <https://github.com/balabit/syslog-ng/pull/885#discussion_r50842063>:
>
> > +
> > +/* This version of strchr() is optimized for cases where the string we are
> > + * looking up characters in is often zero or one character in length.  In
> > + * those cases we can avoid the strchr() call, which can make a tight loop
> > + * doing a lot of strchr() calls a lot faster, especially as this function
> > + * is inlined.
> > + *
> > + * Naming: I originally wanted to use the original function as a prefix
> > + * (strchr), however that would potentially pollute the namespace of the
> > + * library that we are optimizing.  So I've added an underscore prefix in
> > + * order not to clash, with that it is still obvious that it intends to
> > + * behave the same as strchr().
> > + *
> > + * NOTE: don't use this unless strchr() really shows up in your profile.
> > + */
> > +static inline char *
>
> Why not return const char*?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/885/files#r50842063>.
>